code_generation/producer.py

```diff
@@ -118,24 +118,28 @@ def __init__(self, name, call, inputs, output, scopes, subproducers):
         self.output = output
         self.producers = subproducers
         self.scopes = scopes
+        # this can be the case for a recurive usage of a ProducerGroup, in this case, we want to use an internal quantity as output for the group
+        if output == None:
+            self.output = q.Quantity(
+                "PG_internal_quantity_%i" % self.__class__.PG_count
+            )
+            self.__class__.PG_count += 1
         # If call is provided, this is supposed to consume output of subproducers. Creating these internal products below:
         if self.call != None:
             for producer in self.producers:
-                # check that output quantities of subproducers are not yet filled
-                if producer.output != None:
-                    print("Output of subproducers must be None!")
-                    raise Exception
+                # if the subproducer does not have an output quantity, we assign an internally tracked quantity
+                if producer.output == None:
+                    # create quantities that are produced by subproducers and then collected by the final call of the producer group
+                    producer.output = q.Quantity(
+                        "PG_internal_quantity_%i" % self.__class__.PG_count
```

Comment:
if producer is VectorProducer, this needs to be added to its last subproducer in addition. Maybe we should make sure that it has a call to make sure that the last producer is the internal representation of the producer group.